eox_nelp/api_clients/tests/tests_futurex.py

```diff
@@ -1,182 +1,22 @@
-
-        with self.assertLogs(futurex.__name__, level="ERROR") as logs:
-            response = api_client.make_post("fake/path", data)
-
-        self.assertDictEqual(response, {})
-        requests_mock.Session.return_value.post.assert_called_with(
-            url=f"{api_client.base_url}/fake/path",
-            json=data,
-        )
-        self.assertEqual(logs.output, [
-            f"ERROR:{futurex.__name__}:{log_error}"
-        ])
-
-    @patch("eox_nelp.api_clients.futurex.requests")
-    @patch.object(FuturexApiClient, "_authenticate")
-    def test_successful_get(self, auth_mock, requests_mock):
-        """Test case when a GET request success.
-
-        Expected behavior:
-            - Response is the expected value
-            - GET was called with the given data and right url.
-        """
-        auth_mock.return_value = {}
-        response = Mock()
-        response.status_code = 200
-        expected_value = {
-            "status": {"success": True, "message": "successful", "code": 1}
-        }
-        response.json.return_value = expected_value
-        requests_mock.Session.return_value.get.return_value = response
-        params = {"format": "json"}
-        api_client = FuturexApiClient()
-
-        response = api_client.make_get("field-options/vocabulary/language", params)
-
-        self.assertDictEqual(response, expected_value)
-        requests_mock.Session.return_value.get.assert_called_with(
-            url=f"{api_client.base_url}/field-options/vocabulary/language",
-            params=params,
-        )
-
-    @patch("eox_nelp.api_clients.futurex.requests")
-    @patch.object(FuturexApiClient, "_authenticate")
-    def test_failed_get(self, auth_mock, requests_mock):
-        """Test case when a GET request fails.
-
-        Expected behavior:
-            - Response is an empty dict
-            - GET was called with the given data and right url.
-            - Error was logged.
-        """
-        auth_mock.return_value = {}
-        response = Mock()
-        response.status_code = 404
-        response.json.return_value = {"test": True}
-        requests_mock.Session.return_value.get.return_value = response
-        params = {"format": "json"}
-        log_error = (
-            "An error has occurred trying to make a get request to https://testing.com/fake/path with status code 404"
-        )
-        api_client = FuturexApiClient()
-
-        with self.assertLogs(futurex.__name__, level="ERROR") as logs:
-            response = api_client.make_get("fake/path", params)
-
-        self.assertDictEqual(response, {})
-        requests_mock.Session.return_value.get.assert_called_with(
-            url=f"{api_client.base_url}/fake/path",
-            params=params,
-        )
-        self.assertEqual(logs.output, [
-            f"ERROR:{futurex.__name__}:{log_error}"
-        ])
+    def setUp(self):
+        """Setup common conditions for every test case"""
+        self.api_class = FuturexApiClient
```

Comment:
These lines could be?
https://github.com/eduNEXT/eox-nelp/blob/1e501f2b3b74c966c766675cc5d292aeebfbcbd4/eox_nelp/api_clients/tests/tests_futurex.py#L44
https://github.com/eduNEXT/eox-nelp/blob/1e501f2b3b74c966c766675cc5d292aeebfbcbd4/eox_nelp/api_clients/tests/tests_futurex.py#L64
``` python
api_client = self.api_client
```